### PR TITLE
Use comparer depend on the users OS

### PR DESCRIPTION
**Pull request by Tasteful**
## Bug
Fixes: https://github.com/NuGet/Home/issues/7329
Regression: Yes/No  
If Regression then when did it last work:   
If Regression then how are we preventing it in future:   

## Fix
Details: Use platform dependent comparer to avoid path with different cases missed in the nuget packaging process.

## Testing/Validation
Tests Added: Yes.  
Reason for not adding tests:  N/A
Validation done:  Yes, with unit and manual testing.

### CODE REVIEW

**Comment by jainaashish** [comment]
Thanks for the PR. Can you also add a test for to cover this scenario?

**Comment by Tasteful** [comment]
@jainaashish If you can point me in the correct direction for how I should write the test, or if we have any other test that will test the project reference that I can duplicate/modify to also cover this use case.

**Comment by nkolev92** [comment]
@Tasteful 
https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs is our largest suite with pack target tests. 

That's a good starting point.

**Comment by Tasteful** [comment]
@jainaashish Test added + rebase on dev

**Review by jainaashish — APPROVED** [review]
Thanks for the contribution.

**Review by nkolev92 — APPROVED** [review]
LGTM

File: src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs [hunk 1]
```diff
@@ -743,7 +743,7 @@ private ISet<string> CalculateFilesToExcludeInPack(IPackTaskRequest<IMSBuildItem
                     library => Path.GetFullPath(Path.Combine(
                         Path.GetDirectoryName(assetsFile.PackageSpec.RestoreMetadata.ProjectPath),
                         PathUtility.GetPathWithDirectorySeparator(library.MSBuildProject))),
-                    library => new PackageIdentity(library.Name, library.Version));
+                    library => new PackageIdentity(library.Name, library.Version), PathUtility.GetStringComparerBasedOnOS());
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs]: Can you please put the Comparer in a different line. 

It was a bit tricky to read without an editor.

Comment (nkolev92) [reply src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs]: @NuGet/nuget-client fyi.

**Comment by nkolev92** [comment]
Thanks for the contribution @Tasteful

**Pull request merged**